olive/passes/onnx/nvmo_quantization.py

```diff
@@ -40,60 +33,433 @@ class Precision(StrEnumBase):
-                description="Algorithm of weight only quantization. Support 'RTN' and 'AWQ'.",
+                default_value="AWQ",
+                searchable_values=Categorical(["AWQ"]),
+                description="Algorithm of weight only quantization. Supports 'AWQ'.",
+            ),
+            "calibration": PassConfigParam(
+                type_=NVModelOptQuantization.Calibration,
+                default_value="awq_clip",
+                searchable_values=Categorical(["awq_lite", "awq_clip"]),
+                description="Calibration method for weight only quantization. Supports 'awq_lite' and 'awq_clip'.",
+            ),
+            "tokenizer_dir": PassConfigParam(
+                type_=str,
+                required=True,
+                default_value="",  # Updated default value
+                description="Tokenizer directory for calibration method.",
             ),
-            **deepcopy(_dataloader_config),
         }
 
     def validate_search_point(
-        self, search_point: Dict[str, Any], accelerator_spec: AcceleratorSpec, with_fixed_value: bool = False
+        self,
+        search_point: Dict[str, Any],
+        accelerator_spec: AcceleratorSpec,
+        with_fixed_value: bool = False,
     ) -> bool:
         if with_fixed_value:
             search_point = self.config_at_search_point(search_point or {})
 
-        if search_point["precision"] != NVModelOptQuantization.Precision.INT4 or search_point["algorithm"] not in [
-            NVModelOptQuantization.Algorithm.RTN,
-            NVModelOptQuantization.Algorithm.AWQ,
+        # Validate Precision
+        if search_point.get("precision") != NVModelOptQuantization.Precision.INT4:
+            logger.error("Only INT4 quantization is supported.")
+            return False
+
+        # Validate Algorithm
+        if search_point.get("algorithm") not in [
+            NVModelOptQuantization.Algorithm.AWQ.value,
         ]:
-            logger.error("Only INT4 quantization with RTN and AWQ algorithm is supported.")
+            logger.error("Only 'AWQ' algorithm is supported.")
             return False
 
+        # Validate Calibration
+        if search_point.get("calibration") not in [
+            NVModelOptQuantization.Calibration.AWQ_LITE.value,
+            NVModelOptQuantization.Calibration.AWQ_CLIP.value,
+        ]:
+            logger.error("Calibration method must be either 'awq_lite' or 'awq_clip'.")
+            return False
+
+        # Optional: Validate 'tokenizer_dir' if necessary
+        if not search_point.get("tokenizer_dir"):
+            logger.warning("Tokenizer directory 'tokenizer_dir' is not specified.")
+
         return True
 
+    def initialize_quant_config(self, config: Dict[str, Any]):
+        """
+        Initialize the quantization configuration by setting up dependencies and calibration data.
+        """
+        # Import torch and DataLoader
+        try:
+            import torch
```

Comment:
torch is a dependency of Olive. It should be fine with importing torch.

Reply:
done